source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -160,34 +160,41 @@ def compareEndPoints(self, other, which):
 		# past the start.
 		# Compare to the start (not the end) when collapsed.
 		selfEndPoint, otherEndPoint = which.split("To")
-		if selfEndPoint == "end" and not self.hasNoText:
+		if selfEndPoint == "end" and not self._isCollapsed:
 			selfEndPoint = "start"
-		if otherEndPoint == "End" and not other.hasNoText:
+		if otherEndPoint == "End" and not other._isCollapsed:
 			otherEndPoint = "Start"
 		which = f"{selfEndPoint}To{otherEndPoint}"
 		return super().compareEndPoints(other, which=which)
 
 	def setEndPoint(self, other, which):
-		"""Works around a UIA bug on Windows 10 1803 and later."""
-		# Even when a console textRange's start and end have been moved to the
-		# same position, the console incorrectly reports the end as being
-		# past the start.
+		"""Override of L{textInfos.TextInfo.setEndPoint}.
+		Works around a UIA bug on Windows 10 1803 and later that means we can
+		not trust the "end" endpoint of a collapsed (empty) text range
+		for comparisons.
+		"""
 		selfEndPoint, otherEndPoint = which.split("To")
-		# In this case, there is no need to check selfEndPoint
-		# since it is about to be overwritten in the super call.
-		if otherEndPoint == "End" and not other.hasNoText:
+		# In this case, there is no need to check if self is collapsed
+		# since the point of this method is to change its text range, modifying the "end" endpoint of a collapsed
+		# text range is fine.
+		if otherEndPoint == "End" and not other._isCollapsed:
 			otherEndPoint = "Start"
 		which = f"{selfEndPoint}To{otherEndPoint}"
 		return super().setEndPoint(other, which=which)
 
-	def _get_hasNoText(self):
+	def _get__isCollapsed(self):
```

Comment:
Do we actually benefit from caching in this case? If not, I think I would prefer not to use the auto-property form and just name it `def _isCollapsed(self)`

Reply:
Nope, caching for this property would actually be pretty harmful if the text info is changes during one core cycle.